src/index.ts

```diff
@@ -305,26 +311,42 @@ export const createClaimWithChildInstance = (
       );
       return tx;
     },
-    getClaimingIDsForUser: async (walletID: string): Promise<string[]> => {
+    getClaimingIDsForUser: async (
+      walletID: string
+    ): Promise<IDWithClaimStatus[]> => {
       const claimingToken = await getClaimingToken(
         config.web3Provider,
         config.claimingRegistrarAddress
       );
-      let childDomain: ethers.BigNumber;
-      const claimingIDs: string[] = [];
+      const IDs: IDWithClaimStatus[] = [];
```

Comment:
Improperly named variable, and also kinda unclear. What type of id?
Consider calling it `domainIds`